test/opt/test_utils.cpp

```diff
@@ -55,4 +55,48 @@ TEST(JoinNonDebugInsts, Cases) {
                  "the only remaining string"}));
 }
 
+namespace {
+typedef struct {
+  const char* orig_str_;
+  const char* find_substr_;
+  const char* replace_substr_;
+  const char* expected_str_;
+  bool replace_should_succeed_;
+} SubstringReplacementTestCase;
+}
+using ReplaceSubstringInPlaceTest =
+    ::testing::TestWithParam<SubstringReplacementTestCase>;
+
+TEST_P(ReplaceSubstringInPlaceTest, SubstringReplacement) {
+  auto process = std::string(GetParam().orig_str_);
+  ASSERT_STREQ(GetParam().orig_str_, process.c_str());
```

Comment:
Is this `ASSERT_STREQ` necessary? Seems like testing `std::string` conversion for me...